activerecord/test/cases/insert_all_test.rb

```diff
@@ -768,6 +768,27 @@ def test_upsert_all_updates_using_provided_sql
     assert_equal "written", Book.find(2).status
   end
 
+  def test_upsert_all_updates_using_values_function_on_duplicate_raw_sql
+    skip unless supports_insert_on_duplicate_update? && \
+      current_adapter?(:AbstractMysqlAdapter) && \
+      !mariadb? && \
+      database_version >= "8.0.19"
```

Comment:
I think we can just always run it. You query is valid regardless of the DB version so might as well always run it.

Also test aren't allowed to be skipped on CI, so if the query is MySQL only you need to use:

```ruby
if current_adapter?(:Mysql2Adapter) || current_adapter?(:TrilogyAdapter)
  def test_...
  end
end
```

Reply:
Thank you. I wrapped in the if logic as you suggested. Yesterday, I saw a couple of errors when running it on CI. For example, it fails with SQLite3 with:

```
Error:
InsertAllTest#test_upsert_all_updates_using_values_function_on_duplicate_raw_sql:
ActiveRecord::StatementInvalid: SQLite3::SQLException: near "values": syntax error
```